### PR TITLE
adding tests for createmodelutils functions

### PR DESCRIPTION
**Pull request by piguy314159265**
Some code cleanup and adding assert_that code as well

### CODE REVIEW

**Review by ramnathv — CHANGES_REQUESTED** [review]
These tests look good. I would flip the order of arguments in `getNamesFromOrdered` as I think it makes more sense.

File: R/createmodelutils.R [hunk 1]
```diff
@@ -7,20 +7,14 @@
...
   assertthat::assert_that(length(data_names) >= minimum_fields)
-  name_y_var <- data_names[1]
-  if(use_weights){
-    names_x_var <- data_names[2:length(data_names)-1]
-    name_weights_var <- data_names[length(data_names)]
-  } else {
-    names_x_var <- data_names[2:length(data_names)]
-    name_weights_var <- NULL
+  y <- data_names[1]
+  x <- if (use_weights) data_names[2:length(data_names)-1] else data_names[2:length(data_names)]
```
Comment (ramnathv) [inline R/createmodelutils.R]: If the `if-else` statement does not fit in a single line, it is better to use the longer format.

``` r
if (x) {
  doThis()
} else {
  doThat()
}
```

File: R/createmodelutils.R [hunk 2]
```diff
@@ -7,20 +7,14 @@
...
+  if (y %in% x) {
+    x <- x[x != y]
   }
-  # Make sure that the target variable is not included in the set of
-  # predictor variables. If it is, then remove it from the set.
-  if (name_y_var %in% names_x_var) {
-    names_x_var <- names_x_var[names_x_var != name_y_var]
-  }
-  list(x = names_x_var, y = name_y_var, w = name_weights_var)
+  list(x = x, y = y, w = w)
```
Comment (ramnathv) [inline R/createmodelutils.R]: I like this cleanup of code. Makes this function a lot more readable. :+1:

File: R/createmodelutils.R [hunk 3]
```diff
@@ -5,20 +5,16 @@
...
-  if(use_weights){
-    names_x_var <- data_names[2:length(data_names)-1]
-    name_weights_var <- data_names[length(data_names)]
-  } else {
-    names_x_var <- data_names[2:length(data_names)]
-    name_weights_var <- NULL
+  minimum_fields <- 2 + use_weights
+  assertthat::assert_that(length(data_names) >= minimum_fields)
+  y <- data_names[1]
+  x <- if (use_weights) data_names[2:length(data_names)-1] else data_names[2:length(data_names)]
```
Comment (ramnathv) [inline R/createmodelutils.R]: If the if-else statement does not fit in a single line, it is better to use the longer format.

``` r
if (x) {
  doThis()
} else {
  doThat()
}
```

File: R/createmodelutils.R [hunk 4]
```diff
@@ -5,20 +5,16 @@
...
+  if (y %in% x) {
+    x <- x[x != y]
   }
-  # Make sure that the target variable is not included in the set of
-  # predictor variables. If it is, then remove it from the set.
-  if (name_y_var %in% names_x_var) {
-    names_x_var <- names_x_var[names_x_var != name_y_var]
-  }
-  list(x = names_x_var, y = name_y_var, w = name_weights_var)
+  list(x = x, y = y, w = w)
```
Comment (ramnathv) [inline R/createmodelutils.R]: You can remove lines 14-16 and return

``` r
list(x = x[x! = y], y = y, w = w)
```

File: R/createmodelutils.R [hunk 5]
```diff
@@ -5,20 +5,16 @@
...
 getNamesFromOrdered <- function(use_weights, data_names) {
-  name_y_var <- data_names[1]
-  if(use_weights){
-    names_x_var <- data_names[2:length(data_names)-1]
-    name_weights_var <- data_names[length(data_names)]
-  } else {
-    names_x_var <- data_names[2:length(data_names)]
-    name_weights_var <- NULL
+  minimum_fields <- 2 + use_weights
+  assertthat::assert_that(length(data_names) >= minimum_fields)
```
Comment (ramnathv) [inline R/createmodelutils.R]: Just to play the devil's advocate, what happens when you have a single predictor that is also specified as the weighting variable?

Comment (piguy314159265) [inline R/createmodelutils.R]: That's a fair point. I will instead assert classes on x, y, w which will stop the opportunity that they are null and be more robust in general I think.

File: R/createmodelutils.R (hunk 3 above)
Comment (piguy314159265) [reply R/createmodelutils.R]: 03da2f7 gives this a roundabout fix by shortening the line

File: R/createmodelutils.R (hunk 5 above)
Comment (piguy314159265) [reply R/createmodelutils.R]: Good catch. 2630e2e provides a fix for this

File: R/createmodelutils.R (hunk 4 above)
Comment (piguy314159265) [reply R/createmodelutils.R]: a3afe13 does this

File: R/createmodelutils.R (hunk 1 above)
Comment (piguy314159265) [reply R/createmodelutils.R]: 03da2f7 gives this a roundabout fix by shortening the line

**Comment by piguy314159265** [comment]
83acaab reverses names in getNamesFromOrdered

**Comment by piguy314159265** [comment]
those are the last tests I'm planning on pushing for this PR. I'll do a separate one with the decision tree specific tests

**Pull request merged**